src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/ParameterCapturingService.cs

```diff
@@ -1,81 +1,271 @@
                 return;
             }
 
-            try
+            if (_requests?.Writer.TryWrite(payload) != true)
             {
-                _probeManager = new FunctionProbesManager(new LogEmittingProbes(_logger));
-                _isAvailable = true;
+                if (!IsAvailable())
+                {
+                    _eventSource.ServiceNotAvailable(_notAvailableReason!.Value, _notAvailableDetails);
+                    return;
+                }
+                else
+                {
+                    // The channel is full, which should never happen if dotnet-monitor is properly rate limiting requests.
+                    _eventSource.FailedToCapture(
+                        ParameterCapturingEvents.CapturingFailedReason.TooManyRequests,
+                        ParameterCapturingStrings.TooManyRequestsErrorMessage);
+                }
             }
-            catch
+        }
+
+        private void OnStopMessage(EmptyPayload _)
+        {
+            if (_stopRequests?.Writer.TryWrite(true) != true)
             {
-                // TODO: Log
+                if (!IsAvailable())
+                {
+                    _eventSource.ServiceNotAvailable(_notAvailableReason!.Value, _notAvailableDetails);
+                    return;
+                }
+                else
+                {
+                    // The channel is full which is OK as stop requests aren't tied to a specific operation.
+                }
             }
         }
 
-        public void StopCapturing()
+        private void StartCapturing(StartCapturingParametersPayload request)
         {
-            if (!_isAvailable)
+            if (!IsAvailable())
             {
                 throw new InvalidOperationException();
             }
 
-            _probeManager?.StopCapturing();
+            MethodResolver resolver = new();
+            List<MethodInfo> methods = new(request.Methods.Length);
+            List<MethodDescription> methodsFailedToResolve = new();
+
+            for (int i = 0; i < request.Methods.Length; i++)
+            {
+                MethodDescription methodDescription = request.Methods[i];
+
+                List<MethodInfo> resolvedMethods = resolver.ResolveMethodDescription(methodDescription);
+                if (resolvedMethods.Count == 0)
+                {
+                    methodsFailedToResolve.Add(methodDescription);
+                }
+
+                methods.AddRange(resolvedMethods);
+            }
+
+            if (methodsFailedToResolve.Count > 0)
+            {
+                UnresolvedMethodsExceptions ex = new(methodsFailedToResolve);
+                _logger!.LogWarning(ex.Message);
+                throw ex;
+            }
+
+            _probeManager!.StartCapturing(methods);
+            _logger!.LogInformation(
+                ParameterCapturingStrings.StartParameterCapturingFormatString,
+                request.Duration,
+                request.Methods.Length);
```

Comment:
Should this be `request.Methods.Length` or `methods.length`? In the event that there are multiple `resolvedMethods` for a single `request.Method`